frontend/src/app/general/nav-bar-menu/nav-bar-menu.component.html

```diff
@@ -4,34 +4,36 @@
  -->
 
 <mat-list>
-  @for (item of navBarService.navBarItems; track $index) {
-    <div>
-      @if (userService.validateUserRole(item.requiredRole)) {
-        @if (item.href) {
-          <a
-            class="items-center"
-            mat-list-item
-            (click)="navBarService.toggle()"
-            [attr.href]="item.href"
-            [attr.target]="item.target"
-          >
-            {{ item.name }}
-            @if (item.icon) {
-              <mat-icon>{{ item.icon }}</mat-icon>
-            }
-          </a>
-        } @else {
-          <a
-            mat-list-item
-            (click)="navBarService.toggle()"
-            [routerLink]="item.routerLink"
-          >
-            {{ item.name }}
-          </a>
+  @if (navBarService.navbarItems$ | async) {
```

Comment:
We don't need this check. The return value of the `navbarItems$` subscription is never undefined, it can only return a list. Therefore, the for loop will just skip it if empty.